Sandbox/src/Game/Commands/AutoAttackCommand.h

```diff
@@ -0,0 +1,124 @@
+#pragma once
+
+#include "Game/Commands/ICommand.h"
+#include "Game/Combat/MeleeFormulas.h"
+#include "Game/Combat/RangedFormulas.h"
+#include "Game/Combat/MagicFormulas.h"
+#include "Game/Combat/CombatExperience.h"
+
+class AutoAttackCommand : public ICommand
+{
+private:
+	Player* mPlayer;
+	NpcFighter* mNpc;
+
+	inline bool HasAmmo() const
+	{
+		bool hasAmmo = true;
+
+		if (mPlayer->CombatStance() == CombatOption::ranged_accurate ||
+			mPlayer->CombatStance() == CombatOption::ranged_rapid ||
+			mPlayer->CombatStance() == CombatOption::ranged_longrange)
+		{
+			if (Weapon* weapon = dynamic_cast<Weapon*>(mPlayer->Gear().GetItem(Gear::EquipmentSlot::weapon)))
+			{
+				if (weapon->Type() == 1)
+				{
+					hasAmmo = false;
+					for (const auto& ammo : weapon->AmmoIndex())
+					{
+						if (mPlayer->Gear().HasItem(ammo))
+						{
+							hasAmmo = true;
+							break;
+						}
+					}
+				}
+			}
+		}
+
+		return hasAmmo;
+	}
+
+	inline bool HasRunes() const
+	{
+		bool hasRunes = true;
+
+		if (mPlayer->CombatStance() == CombatOption::magic_standard ||
+			mPlayer->CombatStance() == CombatOption::magic_defensive)
+		{
+			if (mPlayer->SpellBook().ActiveSpell() != -1)
+				hasRunes = mPlayer->Inventory().HasItems(mPlayer->SpellBook().Spells()[mPlayer->SpellBook().ActiveSpell()].CastReq());
+		}
+
+		return hasRunes;
```

Comment:
nit: I would've just done something like this:
```
bool hasCombatStance = (mPlayer->CombatStance() == CombatOption::magic_standard || mPlayer->CombatStance() == CombatOption::magic_defensive);
bool hasActiveSpell = (mPlayer->SpellBook().ActiveSpell() != -1);
bool hasRuneItems = mPlayer->Inventory().HasItems(mPlayer->SpellBook().Spells()[mPlayer->SpellBook().ActiveSpell()].CastReq());

return hasCombatStance && hasActiveSpell && hasRunItems;
```
(note: with the variable names changed to whatever they represent cause I'm not completely sure lol)

Reply:
Dang, that's solid. This is why they pay you the big bucks